wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -773,12 +789,31 @@ public void onCreate() {
 
         peerDiscoveryList.add(dnsDiscovery);
         updateAppWidget();
+
+        initViewModel();
+    }
+
+    void initViewModel() {
+
         AppDatabase.getAppDatabase().blockchainStateDao().load().observe(this, new Observer<BlockchainState>() {
             @Override
             public void onChanged(BlockchainState blockchainState) {
                 handleBlockchainStateNotification(blockchainState);
             }
         });
+
+        Wallet wallet = ((WalletApplication) getApplication()).getWallet();
+        //wallet.addCreditFundingEventListener(Threading.SAME_THREAD, creditFundingTransactionEventListener);
```

Comment:
For some reason the listener added here was being called three times.  This resulted in three services being created.